pkg/controller/operators/operatorcondition_controller.go

```diff
@@ -0,0 +1,237 @@
+package operators
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	operatorsv1 "github.com/operator-framework/api/pkg/operators/v1"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/ownerutil"
+)
+
+const (
+	OperatorConditionEnvVarKey = "OPERATOR_CONDITION_NAME"
```

Comment:
Just to clarify - there's only one operator condition per operator, but multiple deployments/serviceaccounts have permission to write to it?

Reply:
At this time yes. In the future, OLM can be updated to allow operator authors to define the deployments and serviceAccounts that should be able to update the resource by providing the operatorcondition in the bundle and specifying said resources in the spec.